src/libratbag.c

```diff
@@ -1194,6 +1204,13 @@ ratbag_resolution_set_active(struct ratbag_resolution *resolution)
 	struct ratbag_profile *profile = resolution->profile;
 	struct ratbag_resolution *res;
 
+	log_debug(profile->device->ratbag,"Setting active resolution index to %d (libratbag)\n",resolution->index);
+	if (resolution->dpi_x == 0) { /* if one axis is zero both have to be zero */
+		/* device would repsond with ERR_INVALID_ARGUMENT */
+		log_error(profile->device->ratbag,"Setting the active resolution to zero dpi is not allowed\n");
+		return -EINVAL;
+	}
```

Comment:
Please remember this is common area, the code is shared between all drivers, so the reference to `ERR_INVALID_ARGUMENT` doesn't make much sense.

I am not sure about the need for this check.

Reply:
We could remove it, since the user shouldn't do it, but in the end it's kinda unrealistic to do it anyway. If piper checks this, we can remove the checks.